File: Editors/DisabledEditorsColor/disablededitorscs-zip/DisabledEditorsCS/DisabledEditors/MyRadSpinEditor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Telerik.WinControls.UI;

namespace DisabledEditors
{
    public class MyRadSpinEditor : RadSpinEditor
    {
        RadTextBoxItem realTB;

        public override string ThemeClassName
        {
            get
            {
                return typeof(RadSpinEditor).FullName;
            }
        }

        MyTextBox mtb = new MyTextBox();
        RadTextBoxItem tbi;

        protected override void CreateChildItems(Telerik.WinControls.RadElement parent)
        {
            base.CreateChildItems(parent);

            realTB = this.SpinElement.TextBoxItem;
            tbi = new RadTextBoxItem(mtb);
            mtb.Font = this.Font;
        }

        protected override void OnEnabledChanged(EventArgs e)
        {
            if (!this.Enabled)
            {
                tbi.Text = this.Text;
                this.SpinElement.Children[2].Children.Remove(realTB);
                this.SpinElement.Children[2].Children.Add(tbi);
            }
            else
            {
                this.SpinElement.Children[2].Children.Add(realTB);
                this.SpinElement.Children[2].Children.Remove(tbi);
            }

            base.OnEnabledChanged(e);
        }
    }
}

File: Editors/DisabledEditorsColor/disablededitorsvb-zip/DisabledEditorsVB/DisabledEditors/MyRadSpinEditor.vb
Imports System.Text
Imports Telerik.WinControls.UI

Namespace DisabledEditors
	Public Class MyRadSpinEditor
        Inherits RadSpinEditor

		Private realTB As RadTextBoxItem
        Private mtb As MyTextBox
		Private tbi As RadTextBoxItem

		Protected Overrides Sub CreateChildItems(ByVal parent As Telerik.WinControls.RadElement)
			MyBase.CreateChildItems(parent)

            mtb = New MyTextBox()
			realTB = Me.SpinElement.TextBoxItem
			tbi = New RadTextBoxItem(mtb)
			mtb.Font = Me.Font
		End Sub

		Protected Overrides Sub OnEnabledChanged(ByVal e As EventArgs)
			If Not Me.Enabled Then
				tbi.Text = Me.Text
				Me.SpinElement.Children(2).Children.Remove(realTB)
				Me.SpinElement.Children(2).Children.Add(tbi)
			Else
				Me.SpinElement.Children(2).Children.Add(realTB)
				Me.SpinElement.Children(2).Children.Remove(tbi)
			End If

			MyBase.OnEnabledChanged(e)
        End Sub

        Public Overrides Property ThemeClassName() As String
            Get
                Return GetType(RadSpinEditor).FullName
            End Get
            Set(ByVal value As String)

            End Set
        End Property
	End Class
End Namespace
